nsa/demo.py

```
@@ -0,0 +1,230 @@
+    print("Running inference...")
+    with torch.no_grad():
+        output = model(img_tensor)
+        pred = output.argmax(dim=1).squeeze().cpu().numpy()
+
+    # Calculate IoU with ground truth
+    gt = pupil_mask.astype(np.int64)
+    intersection = np.logical_and(pred == 1, gt == 1).sum()
+    union = np.logical_or(pred == 1, gt == 1).sum()
+    iou = intersection / max(union, 1)
+
+    print(f"\nResults:")
```

Ruff (0.14.8)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)